insights/v1alpha1/0000_10_01_datagather.crd.yaml

```diff
@@ -0,0 +1,242 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    api-approved.openshift.io: https://github.com/openshift/api/pull/1365
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+  name: datagathers.insights.openshift.io
+spec:
+  group: insights.openshift.io
+  names:
+    kind: DataGather
+    listKind: DataGatherList
+    plural: datagathers
+    singular: datagather
+  scope: Namespaced
```

Comment:
I am not sure whether this would be better as namespaced or clusterscoped.

Reply:
Good question, what do we expect the usage pattern of this to be? If you have it namespaced, will you reconcile objects across all namespaces or just a single pre-defined namespace?

CC @deads2k interested to see what you think about this

One thought, given we are discussing this API owning pods/jobs, having it namespaced then pairs nicely with it being a job semantic. The pod would be created in the same namespace i guess

Reply:
Let me put it this way. I think the periodic gathering will create the CR as well as the job in the `openshift-insights` namespace. The question is more about the on-demand gathering where the CR will be created by the user. I think it would make sense to watch only CRs created in one namespace (`openshift-insights`), because the jobs will be created there anyways...but I don't have so much experience in this area.